shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+ * forwarding some calls from an {@code Activity}, as well as forwarding lifecycle calls from
+ * an {@code Activity} or a {@code Fragment}.
+ *
+ * @see FlutterActivity
+ *
+ * @see io.flutter.view.FlutterView
+ */
+@SuppressWarnings("WeakerAccess")
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1)
+public class FlutterFragment extends Fragment implements PluginRegistry {
+  private static final String TAG = "FlutterFragment";
+
+  private static final String ARG_IS_SPLASH_SCREEN_DESIRED = "show_splash_screen";
+  private static final String ARG_INITIAL_ROUTE = "initial_route";
+  private static final String ARG_APP_BUNDLE_PATH = "app_bundle_path";
+
+  private static final WindowManager.LayoutParams MATCH_PARENT =
+      new WindowManager.LayoutParams(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
+
+  public static FlutterFragment newInstance(boolean isSplashScreenDesired,
+                                            @Nullable String initialRoute,
+                                            @NonNull String appBundlePath) {
+    FlutterFragment frag = new FlutterFragment();
+
+    Bundle args = new Bundle();
+    args.putBoolean(ARG_IS_SPLASH_SCREEN_DESIRED, isSplashScreenDesired);
+    args.putString(ARG_INITIAL_ROUTE, initialRoute);
+    args.putString(ARG_APP_BUNDLE_PATH, appBundlePath);
+    frag.setArguments(args);
+
+    return frag;
+  }
+
+  private FrameLayout container;
+  private FlutterView flutterView;
+  private View launchView;
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
+    createLayout();
+    // TODO: Should we start running the FlutterView here or when attached? It was being done here
+    //       in the Activity, but maybe that was a problem to begin with?
+    doInitialFlutterViewRun();
+
+    return container;
+  }
+
+  @Override
+  public void onStart() {
+    super.onStart();
+    flutterView.onStart();
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    // TODO: should flutterView have an onResume() method?
+  }
+
+  public void onPostResume() {
+    flutterView.onPostResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    flutterView.onPause();
+  }
+
+  @Override
+  public void onStop() {
+    super.onStop();
+    flutterView.onStop();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+
+    // TODO(mattcarroll): re-evaluate how Flutter plugins interact with FlutterView and FlutterNativeView
+    final boolean detach = flutterView.getPluginRegistry().onViewDestroy(
+      flutterView.getFlutterNativeView()
+    );
+    if (detach || retainFlutterNativeView()) {
+      // Detach, but do not destroy the FlutterView if a plugin expressed interest in its
+      // FlutterNativeView.
+      flutterView.detach();
+    } else {
+      flutterView.destroy();
+    }
+  }
+
+  public void onBackPressed() {
```

Comment:
Documentation?

I think we want to support the standalone FlutterFragment use case. So an example and documentation would be helpful.

Reply:
Done